let DEBUG_MODE,ValidationError,CustomErrorHandler;_093‍.x([["default",()=>_093‍.o]]);_093‍.w("../config",[["DEBUG_MODE",["DEBUG_MODE"],function(v){DEBUG_MODE=v}]]);_093‍.w("joi",[["ValidationError",["ValidationError"],function(v){ValidationError=v}]]);_093‍.w("../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);



const errorHandler = (err, req, res, next) => {
    
    let statusCode = 500;
    console.log("Erro is here")

    let data = {
        message: 'Internal server error',
        responseCode: statusCode,
        ...(DEBUG_MODE === 'true' && { originalError: err.message })
    }

    if (err instanceof ValidationError) {
        statusCode = 422;
        data = {
            message: err.message,
            responseCode: statusCode
        }
    }

    if (err instanceof CustomErrorHandler) {
        statusCode = err.status;
        data = {
            message: err.message,
            responseCode: statusCode
        }
    }

    return res.status(statusCode).json(data);
}

_093‍.d(errorHandler);